Frontend/ui-driver/src/Flow.purs

```diff
@@ -2695,7 +2723,10 @@ notificationFlow = do
       modifyScreenState $ NotificationsScreenStateType (\notificationScreen -> state{offsetValue = (length state.notificationList)})
       notificationFlow
     GO_HOME_SCREEN -> homeScreenFlow
-    GO_REFERRAL_SCREEN -> referralScreenFlow
+    GO_REFERRAL_SCREEN -> do
+      let cityConfig = getCityConfig config.cityConfig (getValueToLocalStore DRIVER_LOCATION)
```

Comment:
create a new function `referralFlow`. Inside that check the config and direct the flow.